File: February 2026 GFG SOLUTION/February-18.c
int merge(int arr[], int temp[], int l, int r) {
    if (l >= r) return 0;
    int m = (l + r) / 2, cnt = 0;
    cnt += merge(arr, temp, l, m);
    cnt += merge(arr, temp, m + 1, r);
    int i = l, j = m + 1, k = l;
    while (i <= m && j <= r)
        arr[i] <= arr[j] ? (temp[k++] = arr[i++]) : (cnt += m - i + 1, temp[k++] = arr[j++]);
    while (i <= m) temp[k++] = arr[i++];
    while (j <= r) temp[k++] = arr[j++];
    for (i = l; i <= r; i++) arr[i] = temp[i];
    return cnt;
}

int countInversions(int arr[], int n) {
    int temp[n];
    return merge(arr, temp, 0, n - 1);
}

File: February 2026 GFG SOLUTION/February-18.java
class Solution {
    static int merge(int[] arr, int l, int m, int r) {
        int[] t = Arrays.copyOfRange(arr, l, r + 1);
        int i = 0, j = m - l + 1, k = l, cnt = 0, lsz = m - l + 1;
        while (i < lsz && j < t.length)
            if (t[i] <= t[j]) arr[k++] = t[i++];
            else { cnt += lsz - i; arr[k++] = t[j++]; }
        while (i < lsz) arr[k++] = t[i++];
        while (j < t.length) arr[k++] = t[j++];
        return cnt;
    }
    static int sort(int[] arr, int l, int r) {
        if (l >= r) return 0;
        int m = (l + r) / 2;
        return sort(arr, l, m) + sort(arr, m + 1, r) + merge(arr, l, m, r);
    }
    static int inversionCount(int[] arr) { return sort(arr, 0, arr.length - 1); }
}
